core/src/main/java/io/smallrye/openapi/runtime/io/operation/OperationReader.java

```diff
@@ -106,7 +107,27 @@ public static Operation readOperation(final JsonNode node) {
     }
 
     // Helpers for scanner classes
-    public static boolean operationIsHidden(final MethodInfo method) {
+    public static boolean operationIsHidden(final AnnotationScannerContext context, final MethodInfo method) {
+        if (!context.getConfig().getScanExcludeAnnotations().isEmpty()) {
+            for (DotName scanExcludeAnnotation : context.getConfig().getScanExcludeAnnotations()) {
+                if (method.hasAnnotation(scanExcludeAnnotation)) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        if (!context.getConfig().getScanAnnotations().isEmpty()) {
+            for (DotName scanAnnotation : context.getConfig().getScanAnnotations()) {
+                if (method.hasAnnotation(scanAnnotation)) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
Not quite happy how these inclusion / exlusion checks turned out, but did not find a cleaner way.